crates/wac-resolver/src/registry.rs

```diff
@@ -145,32 +152,33 @@ impl RegistryPackageResolver {
     async fn fetch<'a>(
         &self,
         keys: &IndexMap<BorrowedPackageKey<'a>, SourceSpan>,
-    ) -> Result<(), Error> {
+    ) -> Result<(), CommandError> {
         // First check if we already have the packages in client storage.
         // If not, we'll fetch the logs from the registry.
         let mut fetch = IndexMap::new();
         for (key, span) in keys {
-            let id =
-                key.name
-                    .parse()
-                    .map_err(|e: anyhow::Error| Error::PackageResolutionFailure {
-                        name: key.name.to_string(),
-                        span: *span,
-                        source: e,
-                    })?;
+            let id: PackageName = key.name.parse()?;
 
             // Load the package from client storage to see if we already
             // have a matching version present.
             if let Some(info) = self
                 .client
                 .registry()
-                .load_package(self.client.get_warg_registry(), &id)
-                .await
-                .map_err(|e| Error::PackageResolutionFailure {
-                    name: key.name.to_string(),
-                    span: *span,
-                    source: e,
-                })?
+                .load_package(
+                    self.client
+                        .get_warg_registry(id.namespace())
+                        .await
+                        .map_err(|e| {
+                            CommandError::WargClient(
```

Comment:
It seems like adding an `Error::WargRegistryResolutionError` (or similar) would be a better solution rather than having a `CommandError` type.